### PR TITLE
Mid 82 therapy change request

### PR DESCRIPTION
**Pull request by BigTako**
ticket: https://trello.com/c/C9PB5ZPT/82-%D0%B0%D0%BF-change-request-%D0%B4%D0%BE-%D0%BF%D0%BE%D0%BB%D1%8F-%D1%82%D0%B8%D0%BF-%D1%82%D0%B5%D1%80%D0%B0%D0%BF%D1%96%D1%97
deploy: https://mayak-rouge.vercel.app (login: admin, password: admin)

### CODE REVIEW

File: src/app/(admin)/admin/_utils/transformSpecialistData.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { toConnectList, transformCreateData } from '@admin/_utils/common';
+
+export const transformSpecialistData = ({ specializations, ...rest }) => {
+  // console.log({ rest });
```
Comment (alex-popov-tech) [inline src/app/(admin)/admin/_utils/transformSpecialistData.js]: 🟢 missed this one

File: src/app/api/admin/[...slug]/route.js [hunk 2]
```diff
@@ -1,44 +1,53 @@
...
+    if (!req.auth) throw new NotAuthorizedException();
 
-  const modelName = json.resource.toLowerCase();
-  const isServiceProvider = modelName === 'specialist' || modelName === 'organization';
+    const json = await req.json();
 
-  const getOneTransform = instance => {
-    transformServiceProvider(instance, modelName);
-  };
+    const modelName = json.resource.charAt(0).toLowerCase() + json.resource.slice(1);
```
Comment (alex-popov-tech) [inline src/app/api/admin/[...slug]/route.js]: 🟢  consider using _.camelCase maybe? :)

File: src/app/(admin)/admin/_utils/transformSpecialistData.js (hunk 1 above)
Comment (BigTako) [reply src/app/(admin)/admin/_utils/transformSpecialistData.js]: Thanks for comment! Removed it.

File: src/app/api/admin/[...slug]/route.js (hunk 2 above)
Comment (BigTako) [reply src/app/api/admin/[...slug]/route.js]: Thanks for comment! Discussed in private conversation.

**Review by dmitrenko-v — APPROVED** [review]

File: src/app/(admin)/admin/_components/ServiceProvider/TherapiesCutsSelect.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        validate={required()}
+        fullWidth
+      >
+        <AutocompleteInput
+          isLoading={loading}
+          optionText="title"
+          optionValue="id"
+          label="Тип терапії"
+          validate={required()}
+          onChange={(_, record) => {
```
Comment (cherkasovnew) [inline src/app/(admin)/admin/_components/ServiceProvider/TherapiesCutsSelect.js]: Please move this function to a separate const and use the `useCallback` hook

Comment (BigTako) [reply src/app/(admin)/admin/_components/ServiceProvider/TherapiesCutsSelect.js]: Thanks for comment! Moved to separate useCallback function.

**Review by cherkasovnew — APPROVED** [review]

**Review by alex-popov-tech — APPROVED** [review]

**Pull request merged**